src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,195 @@
+        int numberOfArcherBarbarians = (int) (ARCHER_BARBARIANS_MULTIPLIER * level);
+        int numberOfChiefBarbarians = (int) (CHIEF_BARBARIANS_MULTIPLIER * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > PREFERRED_MAX_HORDE_SIZE && MAX_SIZE == PREFERRED_MAX_HORDE_SIZE && hordeTotal > MAX_SIZE)
+        {
+            //set the preferred horde style if the total spawns is greater that the config's max size
+            numberOfBarbarians = PREFERRED_MAX_BARBARIANS;
+            numberOfArcherBarbarians = PREFERRED_MAX_ARCHERS;
+            numberOfChiefBarbarians = PREFERRED_MAX_CHIEFS;
+        }
+        else if (hordeTotal > MAX_SIZE)
+        {
+            //Equilize the spawns so that there is less spawns than the config's max size
+            numberOfBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfBarbarians);
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            numberOfArcherBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfArcherBarbarians);
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            numberOfChiefBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfChiefBarbarians);
+        }
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, colony, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, colony, raidingWorld);
+    }
+
+    private static int equilizeBarbarianSpawns(final int total,final int numberOf)
+    {
+        int returnValue = numberOf;
+        if (total > Configurations.maxBarbarianHordeSize)
+        {
+            returnValue = total - MAX_SIZE;
+
+            if (returnValue < 0)
+            {
+                return 0;
+            }
+            return returnValue;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Spawns EntityToSPawn at the X, Y, Z
+     */
+
+    public static void spawn(final ResourceLocation entityToSpawn, final int numberOfSpawns, final Colony colony, final World world)
+    {
+
+        final BlockPos targetSpawnPoint = calculateSpawnLocation(world, colony);
+
+        if (targetSpawnPoint != null && entityToSpawn != null && world != null)
+        {
+            final int x = targetSpawnPoint.getX();
+            final int y = targetSpawnPoint.getY();
+            final int z = targetSpawnPoint.getZ();
+
+            IntStream.range(0, numberOfSpawns).forEach(theInteger ->
+            {
+                final Entity entity = EntityList.createEntityByIDFromName(entityToSpawn, world);
+
+                if (entity != null)
+                {
+                    if (entityToSpawn.equals(barbarian))
```

Comment:
I think this whole part here can be extract in a "equipBabarians" method to reduce the lambda size